internal/pkg/term/syncbuffer/termprinter.go

```diff
@@ -0,0 +1,184 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package syncbuffer provides a goroutine safe bytes.Buffer as well printing functionality to the terminal.
+package syncbuffer
+
+import (
+	"fmt"
+	"io"
+	"math"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/aws/copilot-cli/internal/pkg/term/cursor"
+	"golang.org/x/term"
+)
+
+// pollIntervalForSyncBuffer is the time Interval to wait between checking whether the output buffers are done.
+const pollIntervalForSyncBuffer = 60 * time.Millisecond
+
+// FileWriter is the interface to write to a file.
+type FileWriter interface {
+	io.Writer
```

Comment:
Don't we need `Fd() uintptr` as well for this interface? that's what makes it a _File_Writer (the fact that there is a method for a file descriptor)

Reply:
But the `log.DiagnosticWriter` does not implement the below interface if we have `Fd() uintptr`. What can we do in that case 🤔 

```yaml
type FileWriter interface {
   io.Writer
   Fd() uintptr
}
```

Reply:
:cool-crying: we can pass `os.Stderr` then instead of `log.DiagnosticWriter` (they're the same)